src/code_8057824_1.c

```diff
@@ -441,6 +493,516 @@ extern u32 sub_803D73C(u32);
 
 extern void sub_80460F8(Position *, Item *, u8);
 
+extern u8 sub_8044B28(void);
+extern u8 gUnknown_202E038[];
+extern void sub_8085374();
+extern void sub_8068FE0(Entity *, u32, u32);
+extern void DisplayDungeonDialogue(const u8 *);
+extern void sub_8086448(void);
+extern void sub_8086A54(Entity *);
+extern void sub_8087144();
+extern bool8 sub_8085B80(struct_8085B80 *);
+
+void SkarmoryEntry(Entity *);
+
+void sub_8086A3C(Entity *pokemon)
+{
+    pokemon->info->unk15C = 1;
+    pokemon->info->unk15E = 1;
+}
+
+void sub_8086A54(Entity *pokemon)
+{
+    pokemon->info->unk15C = 1;
+    pokemon->info->unk15E = 0;
+}
+
+void SetupBossFightHP(Entity *pokemon, s32 newHP, u16 songIndex)
+{
+
+  EntityInfo *entityInfo;
+
+  entityInfo = GetEntInfo(pokemon);
+
+  entityInfo->bossFlag = TRUE;
+
+  // BUG: Source of the Reviver Seed Boss Glitch
+  //
+  // Video to demonstration:
+  // https://www.youtube.com/watch?v=rHu7EehrZ68
+  entityInfo->originalHP = entityInfo->maxHPStat;
+  if (newHP != 0) {
+    entityInfo->maxHPStat = newHP;
+    entityInfo->HP = newHP;
+  }
+
+  gDungeon->bossSongIndex = songIndex;
+  SetDefaultIQSkills(entityInfo->IQSkillMenuFlags, entityInfo->bossFlag);
+  LoadIQSkills(pokemon);
+}
+
+void sub_8086AC0(void)
+{
+    if(!sub_8044B28())
+        if(gDungeon->unk2 == 0)
+            sub_8097FF8();
+}
+
+u8 sub_8086AE4(s16 _index)
+{
+    s32 pokeIndex = _index;
+
+    if(gDungeon->unk65C == 0)
+        return 1;
+    else
+        return HasRecruitedMon(pokeIndex);
+}
+
+void sub_8086B14(void)
+{
+  Entity * LeaderEntity;
+  Entity * DiglettEntity;
+  Entity * SkarmoryEntity;
+
+  LeaderEntity = xxx_call_GetLeader();
+  DiglettEntity = GetEntityFromClientType(4);
+  SkarmoryEntity = GetEntityFromClientType(3);
+  DungeonStartNewBGM(MUS_IN_THE_DEPTHS_OF_THE_PIT);
+  sub_8085374();
+  sub_80854D4();
+  sub_8085930(DIRECTION_NORTH);
+  sub_80855E4(sub_8086A3C);
+  sub_8086A3C(SkarmoryEntity);
+  DiglettEntity->info->unk15C = 1;
+  sub_8085860(LeaderEntity->pos.x,LeaderEntity->pos.y - 2);
+  CopyMonsterNametoBuffer(gUnknown_202E038,MONSTER_DIGLETT);
+  CopyMonsterNametoBuffer(gUnknown_202E038 + 0x50, MONSTER_SKARMORY);
+}
+
+void sub_8086B94(void)
+{
+  Entity * LeaderEntity;
+  Entity * DiglettEntity;
+  Entity * SkarmoryEntity;
+
+  LeaderEntity = xxx_call_GetLeader();
+  DiglettEntity = GetEntityFromClientType(4);
+  SkarmoryEntity = GetEntityFromClientType(3);
+
+  sub_8068FE0(SkarmoryEntity,0x21c,0);
+  sub_8068FE0(DiglettEntity,0x21c,0);
+  sub_80854D4();
+  sub_8085930(DIRECTION_NORTH);
+  sub_8085860(LeaderEntity->pos.x,LeaderEntity->pos.y);
+}
+
+void sub_8086BDC(char param_1, s32 param_2)
+{
+  if ((((param_2 * 0x1000000) - 0x1000000U) >> 0x18 < 2) && (param_1 == 3)) {
+    sub_8097FA8(1);
+    gDungeon->unk2 = 1;
+  }
+}
+
+void SkarmoryPreFightDialogue(void)
+{
+  Entity *LeaderEntity;
```

Comment:
Nitpick, but I'm used to variables using camelCase, so `LeaderEntity` becomes `leaderEntity`. 👀